clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -528,8 +719,13 @@ void HLSLExternalSemaSource::defineHLSLTypesWithForwardDeclarations() {
                     ResourceKind::TypedBuffer, /*IsROV=*/false,
                     /*RawBuffer=*/true)
         .addArraySubscriptOperators()
+        .addIncrementCounterMethod(*SemaPtr)
```

Comment:
Should we make the BuiltinTypeDeclBuilder store a reference to Sema to avoid needing to pass these pointers around?